packages/api-client-core/src/operationRunners.ts

```diff
@@ -43,28 +67,64 @@ export const findOneByFieldRunner = async <Shape = any>(
   const records = hydrateConnection<Shape>(response, connectionObject);
 
   if (records.length > 1) {
-    throw new GadgetNonUniqueDataError(
-      `More than one record found for ${modelApiIdentifier}.${fieldName} = ${fieldValue}. Please confirm your unique validation is not reporting an error.`
-    );
+    throw getNonUniqueDataError(modelApiIdentifier, fieldName, fieldValue);
   }
 
   return records[0];
 };
 
-export const findManyRunner = async <Shape = any>(
+interface QueryPlan {
+  variables: any;
+  query: string;
+}
+
+const getRecordList = async <Shape = any, Plan extends QueryPlan = QueryPlan>(
   modelManager: AnyModelManager,
   operation: string,
-  defaultSelection: FieldSelection,
-  modelApiIdentifier: string,
+  plan: Plan,
   options?: PaginationOptions
 ) => {
-  const plan = findManyOperation(operation, defaultSelection, modelApiIdentifier, options);
   const response = await modelManager.connection.currentClient.query(plan.query, plan.variables).toPromise();
   const connectionObject = assertOperationSuccess(response, [operation]);
   const records = hydrateConnection<Shape>(response, connectionObject);
   return GadgetRecordList.boot<Shape>(modelManager, records, { options, pageInfo: connectionObject.pageInfo });
 };
 
+export const findManyRunner = async <Shape = any>(
+  modelManager: AnyModelManager,
+  operation: string,
+  defaultSelection: FieldSelection,
+  modelApiIdentifier: string,
+  options?: PaginationOptions
+) => {
+  const plan = findManyOperation(operation, defaultSelection, modelApiIdentifier, options);
+  return await getRecordList<Shape>(modelManager, operation, plan, options);
+};
+
+export const findFirstRunner = async <Shape = any>(
```

Comment:
Looking at it again, I could just force a `first: 1` option into the various places I'm using `findFirstOperation`, to switch to `findManyOperation` instead.

I'll take a closer look at the main repo code gen, but I might be able to do the same thing there, to ditch these extra runners.

Reply:
Oh... the only issue is that `findMany` and `findOne` call `assertOperationSuccess`, which will throw (rather than return) an error if the data is null — but they also throw other errors as well.

That said, I can just add a new flag that can be passed into the `findMain` and `findOne` runners, to let them determine whether to throw the empty error or not; since it would only add to the existing function call signature, it shouldn't break any existing usage.

I think I'll do that then, which will reduce this code surface even more.